### DISCUSSION TITLE
Passing Request in StrategyVerifyCallback

### DISCUSSION THREAD

**Discussion by maou-shonen** [sergiodxa/remix-auth#280] [Ideas]
Is it possible to include the request in the StrategyVerifyCallback? I believe there are quite a few examples where the login verification process between enterprises references headers.
Currently, my solution involves forking the required Strategy and modifying it accordingly.

For example, in FormStrategy:
```ts
export class MyFormStrategy<User> extends Strategy<
  User,
  FormStrategyVerifyParams
> {
  name = 'my-form'

  async authenticate(
    request: Request,
    sessionStorage: SessionStorage,
    options: AuthenticateOptions,
  ): Promise<User> {
+   options.context ??= {}
+   options.context.request = request

    // ...The following section is identical to the original.
```

If the request is passed during the `authenticator.use` stage, there's no need to maintain a separate forked version. It's plug-and-play. Isn't this a good idea?

Note 1: I know we can use Express to LoadContext, but it's obviously much more cumbersome.
Note 2: Remix middleware can also handle this, but it seems there's still quite some time before it becomes stable.

**Comment by sergiodxa** [comment]
This depends on the strategy, the OAuth2Strategy and their based strategies all get the request, if you want to send a PR to the FormStrategy to pass the request I could merge it

**Reply by maou-shonen** [reply]
Thank you for your prompt response. I've submitted a PR: https://github.com/sergiodxa/remix-auth-form/pull/35
